### PR TITLE
Enhance Documentation - Design

### PR DESCRIPTION
**Pull request by naveensrinivasan**
1. Added a visual representation of a Git repository structure.
2. Expanded the definition of 'Actors' in the context of a Git repository and the gittuf system.
3. Introduced a new term 'State' and provided a detailed explanation and example of it.

### CODE REVIEW

File: docs/design-document.md [hunk 1]
```diff
@@ -39,24 +39,56 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
...
+Repository
+|
+|-- refs
+|   |
+|   |-- heads
+|   |   |-- master (refers to commit C)
+|   |   |-- feature-x (refers to commit E)
+|   |
+|   |-- tags
+|   |   |-- v1.0 (refers to commit B)
```
Comment (adityasaky) [inline docs/design-document.md:52]: Can you rename `commits` to be `objects` below and include a `v1.0` tag object that in turn refers to commit `B`?

Comment (adityasaky) [reply docs/design-document.md:52]: this message here would then say `refers to tag v1.0`

File: docs/design-document.md [hunk 2]
```diff
@@ -39,24 +39,56 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
...
+These changes can involve any part of the repository, such as modifying files, branches or tags. 
+In the gittuf system, each actor is identified by a unique signing key that they use to sign their contributions. 
+This means that when a policy allows an actor to make certain changes, 
+it's actually allowing the person who has the specific signing key to make those changes. To maintain security, 
+all actions made in the repository, such as adding or modifying files, are checked for authenticity. 
+This is done by verifying the digital signature attached to the action, 
+which must match the public key associated with the actor who is supposed to have made the change.
 
 ### State
+The term "State" refers to the latest values or conditions of the tracked references
```
Comment (adityasaky) [inline docs/design-document.md:78]:
```suggestion

The term "State" refers to the latest values or conditions of the tracked references 
```

File: docs/design-document.md [hunk 3]
```diff
@@ -39,24 +39,56 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
...
 
-In the context of a Git repository, an actor is any user who contributes changes
-to the repository. This may be to any file tracked by the repository in any Git
-ref. In gittuf, actors are identified by the signing keys they use when
-contributing to the repository. A policy that grants an actor the ability to
-make a certain change in fact grants it to the holder of their signing key.
-Verification of any action performed in the repository depends, among other
-factors, on the successful verification of the action's signature using the
-expected actor's public or verification key.
+In a Git repository, an 'actor' is any user who makes changes to the repository.
```
Comment (adityasaky) [inline docs/design-document.md:68]: nit: can you wrap to 80 cols?

File: docs/design-document.md [hunk 4]
```diff
@@ -39,24 +39,56 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
...
 
 ### State
+The term "State" refers to the latest values or conditions of the tracked references 
+(like branches and tags) in a Git repository.
+These are determined by the most recent entries in the [reference state log](#reference-state-log-rsl).
+
+When analyzing changes within the repository, attention is given specifically to the updates 
+pertinent to each individual reference, ensuring a focused review of changes for each branch or tag separately.
+
+#### Example:
```
Comment (adityasaky) [inline docs/design-document.md:85]:
```suggestion
#### Example

```

File: docs/design-document.md (hunk 3 above)
Comment (adityasaky) [reply docs/design-document.md:68]: also nit: trailing whitespaces in a bunch of these lines.

**Review by adityasaky — CHANGES_REQUESTED** [review]
A few minor nits aside, I think this is great to add to the doc. Thanks!

**Comment by adityasaky** [comment]
@naveensrinivasan I took the liberty of pushing 56fc7f2abb9c7ffa84077cc16d3e176c7793fdee to fix some col length / trailing whitespaces to clarify what my editor was flagging. Could you fetch that commit and squash all the commits (don't worry about co-authored-by etc for this one btw)? Thanks!

**Comment by adityasaky** [comment]
On a side note, we should probably lint the markdown files as well in CI :thinking:

File: docs/design-document.md [hunk 5]
```diff
@@ -39,24 +39,63 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
...
-[reference state log](#reference-state-log-rsl). Note that when inspecting
-changes to the state of the repository, a workflow may only consider state
-updates relevant to a particular ref.
+The term "State" refers to the latest values or conditions of the tracked
+references (like branches and tags) in a Git repository.  These are determined
+by the most recent entries in the [reference state
+log](#reference-state-log-rsl).
+
+When analyzing changes within the repository, attention is given specifically to
+the updates pertinent to each individual reference, ensuring a focused review of
```
Comment (JustinCappos) [inline docs/design-document.md:88]: Can you clarify this sentence?  Especially what you mean by "attention is given specifically to
the updates pertinent to each individual reference,"

File: docs/design-document.md [hunk 6]
```diff
@@ -39,24 +39,63 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
...
+changes for each branch or tag separately.
+
+#### Example
+
+If a Git repository has two branches, `main` and `feature`, the state of the
+main branch might be defined by the latest commit, say Commit A. Similarly, the
```
Comment (JustinCappos) [inline docs/design-document.md:94]: What do you mean by "defined"?

File: docs/design-document.md [hunk 7]
```diff
@@ -39,24 +39,63 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
...
+feature branch's state could be determined by a different commit, Commit B. When
+reviewing the repository's history, changes in the main branch are tracked up to
+Commit A, while for the feature branch, up to Commit B.  This method allows for
```
Comment (JustinCappos) [inline docs/design-document.md:97]: I'm not exactly sure what you're proposing / saying here.  What method do you mean and how does this vary from git?

**Review by JustinCappos — COMMENTED** [review]
Looks pretty good!  Just a few more things need to be clarified.  I'm happy to discuss on slack or elsewhere if my feedback isn't clear.

**Review by JustinCappos — APPROVED** [review]

**Comment by adityasaky** [comment]
@naveensrinivasan I pushed some updates based on Justin's feedback. Thank you for the changes here, I think the actor text clarification + the .git structure in particular is helpful.

**Review by adityasaky — APPROVED** [review]

**Pull request merged**